plugins/linguist-backend/src/plugin.ts

```diff
@@ -19,69 +19,79 @@ import {
 import { HumanDuration } from '@backstage/types';
 
-/**
- * Options for Linguist backend plugin
- *
- * @public
- */
-export interface LinguistPluginOptions {
-  schedule?: TaskScheduleDefinition;
-  age?: HumanDuration;
-  batchSize?: number;
-  useSourceLocation?: boolean;
-  linguistJsOptions?: Record<string, unknown>;
-  kind?: string[];
-}
+import { createRouter } from './service/router';
 
 /**
  * Linguist backend plugin
  *
  * @public
  */
-export const linguistPlugin = createBackendPlugin(
-  (options: LinguistPluginOptions) => ({
-    pluginId: 'linguist',
-    register(env) {
-      env.registerInit({
-        deps: {
-          logger: coreServices.logger,
-          reader: coreServices.urlReader,
-          database: coreServices.database,
-          discovery: coreServices.discovery,
-          scheduler: coreServices.scheduler,
-          tokenManager: coreServices.tokenManager,
-          httpRouter: coreServices.httpRouter,
-        },
-        async init({
-          logger,
-          reader,
-          database,
-          discovery,
-          scheduler,
-          tokenManager,
-          httpRouter,
-        }) {
-          httpRouter.use(
-            await createRouter(
-              {
-                ...options,
-              },
-              {
-                logger: loggerToWinstonLogger(logger),
-                reader,
-                database,
-                discovery,
-                scheduler,
-                tokenManager,
-              },
-            ),
+export const linguistPlugin = createBackendPlugin({
+  pluginId: 'linguist',
+  register(env) {
+    env.registerInit({
+      deps: {
+        logger: coreServices.logger,
+        config: coreServices.rootConfig,
+        reader: coreServices.urlReader,
+        database: coreServices.database,
+        discovery: coreServices.discovery,
+        scheduler: coreServices.scheduler,
+        tokenManager: coreServices.tokenManager,
+        httpRouter: coreServices.httpRouter,
+      },
+      async init({
+        logger,
+        config,
+        reader,
+        database,
+        discovery,
+        scheduler,
+        tokenManager,
+        httpRouter,
+      }) {
+        let schedule;
+        if (config.has('linguist.schedule')) {
+          schedule = readTaskScheduleDefinitionFromConfig(
+            config.getConfig('linguist.schedule'),
           );
-        },
-      });
-    },
-  }),
-);
+        }
+        const batchSize = config.getOptionalNumber('linguist.batchSize');
+        const useSourceLocation = config.getBoolean(
+          'linguist.useSourceLocation',
+        );
+        const age = config.getOptionalConfig('linguist.age') as
+          | HumanDuration
+          | undefined;
+        const kind = config.getOptionalStringArray('linguist.kind');
+        const linguistJsOptions = config.getOptionalConfig(
+          'linguist.linguistJsOptions',
+        );
```

Comment:
This one I'm a little unsure about. We have it as a `JsonObject` in the config so is that what we get back here? We've had some issues with this recently so wanting to be sure this continues to work. 

Here's what these options look like from the LinguistJS project which is what does the language logic: https://github.com/Nixinova/LinguistJS/blob/5b176c62232f62c7015a8ddff153519612824276/src/types.ts#L8

Reply:
adjusted the type - lmk if you would suggest a different change :)